openapi-cli/src/main/java/io/ballerina/openapi/generators/openapi/OpenApiConverter.java

```diff
@@ -120,4 +128,57 @@ public void generateOAS3DefinitionsAllService(Path servicePath, Path outPath, St
             }
         }
     }
+
+    /**
+     * This method use for checking the duplicate files.
+     *
+     * @param outPath     output path for file generated
+     * @param openApiName given file name
+     * @return file name with duplicate number tag
+     */
+    private static String checkDuplicateFiles(Path outPath, String openApiName, Boolean isJson) {
+        if (outPath != null && Files.exists(outPath)) {
+            final File[] listFiles = new File(String.valueOf(outPath)).listFiles();
+            if (listFiles != null) {
+                openApiName = checkAvailabilityOfGivenName(openApiName, listFiles, isJson);
+            }
+        }
+        return openApiName;
+    }
+
+    private static String checkAvailabilityOfGivenName(String openApiName, File[] listFiles, Boolean isJson) {
+        for (File file : listFiles) {
+            if (System.console() != null && file.getName().equals(openApiName)) {
+                String userInput = System.console().readLine("There is already a/an " + file.getName() +
+                        " in the location. Do you want to override the file? [y/N] ");
+                if (!Objects.equals(userInput.toLowerCase(Locale.ENGLISH), "y")) {
+                    int duplicateCount = 0;
+                    openApiName = setGeneratedFileName(listFiles, openApiName, duplicateCount, isJson);
+                }
+            }
+        }
+        return openApiName;
+    }
+
+    /**
+     * This method for setting the file name for generated file.
+     *
+     * @param listFiles      generated files
+     * @param fileName       File name
+     * @param duplicateCount add the tag with duplicate number if file already exist
+     */
+    private static String setGeneratedFileName(File[] listFiles, String fileName, int duplicateCount, boolean isJson) {
```

Comment:
Do we need to pass `duplicateCount` as a parameter? It seems it can be converted to a method-local variable